src/elisp/treemacs-extensions.el

```diff
@@ -138,8 +138,7 @@ Also pass additional DATA to predicate function.")
                 (when (or (null predicate) (funcall predicate workspace))
                   (when (or has-previous (not is-first))
                     (treemacs--insert-root-separator))
-                  (setq is-first nil)
-                  (funcall extension))))
+                  (setq is-first (and is-first (funcall extension))))))
```

Comment:
Variadic extensions may print any number of root-level nodes, including zero. Each individual node will print the separator, so variadic extensions retunr `t` here.